#
# Configuration for testing MPC
#

simu:
  clock_publisher:
    publish_rate: 1000  # frequency (Hz) of publish on the /clock topic (ROS frequency)
    time_factor: 2.0    # time factor of the simulation (1.0 for same as wall time)

  #
  # Parameters changed from robot_simulator
  #
  robot_simulator:
    # Model parameters
    init_state: [0, 0.1, 0,  # x, y, z (position)
                 0, 0, 0,  # phi, theta, psi (orientation)
                 0, 0, 0,  # u, v, w (linear speed)
                 0, 0, 0,  # p, q, r (angular speed)
                 0.0]      # delta_m (ballast)

experiments:
  traj_publisher:
    path_freq: 1.0  # publish frequency of the path
    path_frame: 'ocean'  # frame in which the path is expressed
    plan_res: 0.1        # spatial resolution (m) of the planned trajectory
    wp_file_name: 'resources/mpc_test_wp.txt'  # relative path of the text file containing the waypoints

  experiment_stats:
    main_freq: 10.0  # frequency of the main loop
    out_file_name: '/tmp/mpc_out.csv'  # path of the output file for the results of the test
    save_gp: false    # whether to save the GP mean at the end
    load_gp: false    # whether to load the GP mean at the beginning
    gp_file_name: ''  # path of the saved GP mean
